djangosaml2idp/views.py

```diff
@@ -119,11 +168,20 @@ def login_process(request):
     AUTHN_BROKER = AuthnBroker()
     AUTHN_BROKER.add(authn_context_class_ref(req_authn_context), "")
 
+    def _fetch_name_id_by_settings():
+        specified_name_id_field = sp_config.get('name_id_field', None)
+        if specified_name_id_field:
+            if 'request.user.username' == specified_name_id_field:
```

Comment:
- We return `request.user.username` as NameID if specifying flag `'name_id_field' = 'request.user.username'` for centrical site in settings. 
 - Otherwise, we still return `request.user.id` as NameID.